packages/lib/RotatingLogs.ts

```diff
@@ -0,0 +1,73 @@
+const moment = require('moment');
+const { FsDriverDummy } = require('./fs-driver-dummy.js');
+
+export default class RotatingLogs {
+
+	public static fsDriver_: any = null;
+
+	private logFileDir: string = '';
+	private logFileName: string = 'log.txt';
+	private logFileExpirationTimeInDays: number = 30;
+	private logFileMaximumSizeInBytes: number = 1024 * 1024 * 10;
+
+	public constructor(logFileDir: string) {
+		this.setLogFileDir(logFileDir);
+	}
+
+	public setLogFileDir(value: string) {
+		this.logFileDir = value;
+	}
+
+	private fsDriver() {
+		if (!RotatingLogs.fsDriver_) RotatingLogs.fsDriver_ = new FsDriverDummy();
+		return RotatingLogs.fsDriver_;
+	}
+
+	public async cleanLogFile(): Promise<boolean> {
+		const stats = await this.fsDriver().stat(this.logFileFullpath());
+		const sizeInBytes: number = stats.size;
+		if (this.isLogFileBiggerThanMaximumSizeInBytes(sizeInBytes)) {
+			const newLogFile: string = this.logFileFullpath(this.getNameToNonActiveLogFile());
+			this.fsDriver().move(this.logFileFullpath(), newLogFile);
+			return true;
+		}
+		return false;
+	}
+
+	private isLogFileBiggerThanMaximumSizeInBytes(logFileSizeInBytes: number): boolean {
+		return logFileSizeInBytes >= this.logFileMaximumSizeInBytes;
+	}
```

Comment:
I'd remove this function and just test `logFileSizeInBytes >= this.logFileMaximumSizeInBytes` wherever it's needed. It's good to create function to clarify the code in some cases, but in this case it's not necessary